src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -1,370 +1,432 @@
+            else if (underlyingType == typeof(float)) span = MemoryMarshal.Cast<float, TEnum>(GetEnumInfo<float>(rt, getNames: false).Values);
+            else if (underlyingType == typeof(double)) span = MemoryMarshal.Cast<double, TEnum>(GetEnumInfo<double>(rt, getNames: false).Values);
+#if !NATIVEAOT // this won't ever be used, but it still triggers IL3050
+            else if (underlyingType == typeof(bool)) return (TEnum[])GetValues(typeof(TEnum)); // special-cased to return, as GetValues is already cloning
 #endif
+            else throw CreateUnknownEnumTypeException();
+
+            // Return a clone of the array to avoid exposing the cached array instance.
+            return span.ToArray();
+        }
 
+        /// <summary>Retrieves an array of the values of the constants in a specified enumeration.</summary>
+        /// <param name="enumType">An enumeration type.</param>
+        /// <returns>An array that contains the values of the constants in <paramref name="enumType"/>.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="enumType"/> is null.</exception>
+        /// <exception cref="ArgumentException"><paramref name="enumType"/> is not an <see cref="Enum"/>.</exception>
+        /// <exception cref="InvalidOperationException">
+        /// The method is invoked by reflection in a reflection-only context, or <paramref name="enumType"/> is a type from an assembly loaded in a reflection-only context.
+        /// </exception>
         [RequiresDynamicCode("It might not be possible to create an array of the enum type at runtime. Use the GetValues<TEnum> overload or the GetValuesAsUnderlyingType method instead.")]
         public static Array GetValues(Type enumType)
         {
             ArgumentNullException.ThrowIfNull(enumType);
             return enumType.GetEnumValues();
         }
 
-        /// <summary>
-        /// Retrieves an array of the values of the underlying type constants in a specified enumeration type.
-        /// </summary>
+        /// <summary>Retrieves an array of the values of the underlying type constants in a specified enumeration type.</summary>
         /// <typeparam name="TEnum">An enumeration type.</typeparam>
-        /// /// <remarks>
+        /// <remarks>
         /// You can use this method to get enumeration values when it's hard to create an array of the enumeration type.
         /// For example, you might use this method for the <see cref="T:System.Reflection.MetadataLoadContext" /> enumeration or on a platform where run-time code generation is not available.
         /// </remarks>
         /// <returns>An array that contains the values of the underlying type constants in <typeparamref name="TEnum" />.</returns>
-        public static Array GetValuesAsUnderlyingType<TEnum>() where TEnum : struct, Enum =>
-            typeof(TEnum).GetEnumValuesAsUnderlyingType();
+        public static Array GetValuesAsUnderlyingType<TEnum>() where TEnum : struct, Enum
```

Comment:
I would keep the existing implementation that just returns `typeof(TEnum).GetEnumValuesAsUnderlyingType()` for similar reason.